reinforcement_learning/rlclientlib/utility/data_buffer.cc

```diff
@@ -3,50 +3,43 @@
 #include <iterator>
 #include <iostream>
 
-namespace reinforcement_learning { namespace utility {
-  data_buffer::data_buffer() = default;
+namespace reinforcement_learning {
+  namespace utility {
+    data_buffer::data_buffer() = default;
 
-  data_buffer::data_buffer(const translate_func& _translate) : translate(_translate) {}
+    void data_buffer::reset() { _buffer.clear(); }
 
-  void data_buffer::reset() { _buffer.clear(); }
+    size_t data_buffer::size() const {
+      return _buffer.size();
+    }
 
-  std::string data_buffer::str() const{
-    std::string retval(std::begin(_buffer), std::end(_buffer));
-    return retval;
-  }
-
-  size_t data_buffer::size() const {
-    return _buffer.size();
-  }
-
-  void data_buffer::remove_last() { _buffer.pop_back(); }
+    void data_buffer::remove_last() { _buffer.pop_back(); }
 
-  buffer_factory::buffer_factory(const translate_func& _translate) : translate(_translate) {}
+    void data_buffer::append(const unsigned char* data, size_t len) {
+      _buffer.reserve(_buffer.size() + len);
+      _buffer.insert(_buffer.end(), data, data + len);
```

Comment:
Since we are only reserving as much space as we are adding - is there a perf gain by doing this in two steps?

Reply:
It was reusing the same logic as before, reserve before insert. Might need to go back to history to figure why we had it.